emoji/src/main/java/com/vanniktech/emoji/EmojiVariantPopup.java

```diff
@@ -21,38 +22,80 @@
 
 final class EmojiVariantPopup {
   private static final int MARGIN = 2;
+  private static final int DONT_UPDATE_FLAG = -1;
 
+  @NonNull private final View rootView;
   @Nullable private final OnEmojiClickedListener listener;
 
-  private PopupWindow popupWindow;
+  @Nullable private PopupWindow popupWindow;
 
-  EmojiVariantPopup(@Nullable final OnEmojiClickedListener listener) {
+  EmojiVariantPopup(@NonNull final View rootView, @Nullable final OnEmojiClickedListener listener) {
+    this.rootView = rootView;
     this.listener = listener;
   }
 
   void show(@NonNull final View clickedImage, @NonNull final Emoji emoji) {
     dismiss();
 
-    final View content = View.inflate(clickedImage.getContext(), R.layout.emoji_skin_popup, null);
-    final LinearLayout imageContainer = (LinearLayout) content.findViewById(R.id.container);
+    final View content = initView(clickedImage.getContext(), emoji, clickedImage.getWidth());
 
-    popupWindow = new PopupWindow(content,
-            WindowManager.LayoutParams.WRAP_CONTENT,
-            WindowManager.LayoutParams.WRAP_CONTENT
-    );
+    popupWindow = new PopupWindow(content, WindowManager.LayoutParams.WRAP_CONTENT, WindowManager.LayoutParams.WRAP_CONTENT);
+    popupWindow.setFocusable(true);
+    popupWindow.setOutsideTouchable(true);
+    popupWindow.setInputMethodMode(PopupWindow.INPUT_METHOD_NOT_NEEDED);
+    popupWindow.setBackgroundDrawable(new BitmapDrawable(clickedImage.getContext().getResources(), (Bitmap) null));
+
+    content.measure(makeMeasureSpec(0, View.MeasureSpec.UNSPECIFIED), makeMeasureSpec(0, View.MeasureSpec.UNSPECIFIED));
+
+    final Point location = Utils.locationOnScreen(clickedImage);
+
+    final int x = location.x - content.getMeasuredWidth() / 2 + clickedImage.getWidth() / 2;
+    final int y = location.y - content.getMeasuredHeight();
+
+    popupWindow.showAtLocation(rootView, Gravity.NO_GRAVITY, x, y);
+
+    content.post(new Runnable() {
+      @Override public void run() {
+        final Point actualLocation = Utils.locationOnScreen(content);
+
+        // If the actual location the popup is shown at differs from the wanted (Can happen with
+        // dialogs), we have to fix it.
+        if (!(actualLocation.x == x && actualLocation.y == y)) {
+          final int differenceX = actualLocation.x - x;
+          final int differenceY = actualLocation.y - y;
+
+          final int fixedOffsetX = actualLocation.x > x ? x - differenceX : x + differenceX;
+          final int fixedOffsetY = actualLocation.y > y ? y - differenceY : y + differenceY;
+
+          popupWindow.update(fixedOffsetX, fixedOffsetY, DONT_UPDATE_FLAG, DONT_UPDATE_FLAG);
```

Comment:
This might look weird, but is needed for dialogs. I have not found a way to calculate the offset properly, so we `fix` it here if it differs from the wanted location.